aten/src/ATen/core/Tensor.h

```diff
@@ -280,12 +280,13 @@ class CAFFE2_API Tensor {
   /// TensorOptions.h.
   TensorOptions options() const;
 
-  void* data_ptr() const {
-    return this->unsafeGetTensorImpl()->data();
-  }
+  template <typename T = void>
```

Comment:
Is it not possible to do the old construction, e.g.
```
void *data_ptr() const;

template<typename T>
T * data_ptr() const;
```

that seems a bit nicer in that you can say "the non-templatized function is not checked" rather than "the function is checked, except if the template paramter is void".

Reply:
Fixed